src/array/cuda/segment_reduce.cuh

```diff
@@ -63,6 +63,27 @@ __global__ void ScatterAddKernel(
   }
 }
 
+/*!
+ * \brief CUDA kernel to update gradients for reduce op max/min
+ * \note each blockthread is responsible for adding a row in feature tensor
+ *       to a target row in output tensor.
+ */
+template <typename IdType, typename DType>
+__global__ void UpdateGradMinMaxHeteroKernel(
+    const DType *feat, const IdType *idx, const IdType *idx_type, DType *out,
+    int64_t n, int64_t dim, int type) {
+  for (int row = blockIdx.x; row < n; row += gridDim.x) {
+    int col = blockIdx.y * blockDim.x + threadIdx.x;
+    while (col < dim) {
+      if (type == idx_type[row * dim + col]) {
+        const int write_row = idx[row * dim + col];
```

Comment:
Why do we have a two dimensional index that maps to one dimensional index? Is each row of `idx` identical?

Reply:
Again I wanted to write a similar code to existing DGL codes with 2D thread mapping. Pushed a new commit with 1D mapping.